### PR TITLE
util: support simple like

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @coocood @qiuyesuifeng PTAL

### CODE REVIEW

File: src/util/xeval/evaluator.rs [hunk 1]
```diff
@@ -159,8 +160,30 @@ impl Evaluator {
...
+        if Datum::Null == target || Datum::Null == pattern {
+            return Ok(Datum::Null);
+        }
+        let mut target_str = try!(target.into_string());
+        let mut pattern_str = try!(pattern.into_string());
+        if pattern_str.chars().any(|x| x.is_ascii() && x.is_alphabetic()) {
+            target_str = target_str.to_ascii_lowercase();
+            pattern_str = pattern_str.to_ascii_lowercase();
+        }
+        // for now, tidb ensure that the petterns match %?[^_%]*%? will be push down.
```
Comment (coocood) [inline src/util/xeval/evaluator.rs]: will not be.

**Comment by coocood** [comment]
LGTM

File: src/util/xeval/evaluator.rs [hunk 2]
```diff
@@ -288,6 +321,18 @@ mod test {
 			(not_expr(Datum::I64(1)), Datum::I64(0)),
 			(not_expr(Datum::I64(0)), Datum::I64(1)),
 			(not_expr(Datum::Null), Datum::Null),
+            // like operation
+            (like_expr("a", ""), Datum::I64(0)),
```
Comment (qiuyesuifeng) [inline src/util/xeval/evaluator.rs]: Alignment.

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**